chore(content-drive): Added "New Folder" dialog

KevinDavilaDotCMS opened the pull request:
https://github.com/user-attachments/assets/d32ff03b-a35a-47d7-9726-4b1af211590a



This PR fixes: #33253

This PR fixes: #33253

kevindaviladev committed:
WIP - Prev to create a sidebar feature store

kevindaviladev committed:
Fixed existing tests after changes

kevindaviladev committed:
Added tests to withSidebar and moved folder to reusable models and testing utils

kevindaviladev committed:
Added and fixed tests con dialog folder

kevindaviladev committed:
Added file extensions suggestions. Added and fixed tests and validations on Http Errors

kevindaviladev committed:
Improved Body interface name on DotCreateFolder

rjvelazco reviewed (approved):
Left some comments! Most of them are good to have

rjvelazco commented on core-web/libs/portlets/dot-content-drive/portlet/src/lib/shared/constants.ts:
```diff
@@ -96,3 +96,27 @@ export const PANEL_SCROLL_HEIGHT = '25rem';
+    { id: 'FileAsset', name: 'File' }
+];
+
+export const ALLOWED_FILE_EXTENSIONS = [
+    '*.jpg',
+    '*.jpeg',
+    '*.png',
+    '*.gif',
+    '*.png',
+    '*.csv',
+    '*.xls',
+    '*.xlsx',
+    '*.pdf',
+    '*.doc',
+    '*.docx',
+    '*.txt',
+    '*.zip',
+    '*.rar',
+    '*.tar',
+    '*.gz'
+];
```
Oh, I see if to select what types users want to allow on the new folder!

kevindaviladev committed:
feat(content-drive): enhance folder creation dialog with improved UX and API integration

Key improvements:
•  API Integration: Replace hardcoded file asset types with dynamic fetching from DotContentTypeService
•  User Experience: Add autofocus to title field and loading states during folder creation
•  Code Organization: Move DotFolderEntity interface to shared models and add comprehensive JSDoc documentation
•  Styling: Improve dialog animations, consistent spacing with design tokens, and responsive layout
•  Error Handling: Switch from async/await to reactive patterns with proper error handling
•  Testing: Add comprehensive test coverage for new utility functions and update existing tests
•  Performance: Refactor folder hierarchy fetching with utility functions for better reusability

Technical changes:
•  Moved DotFolderEntity interface from service to models package
•  Added getFolderHierarchyByPath and getFolderNodesByPath utility functions
•  Enhanced form validation and user feedback with loading indicators
•  Improved SCSS with design token usage and smoother animations
•  Updated dialog width to 500px for better content presentation

kevindaviladev committed:
Manually merged with main

kevindaviladev committed:
Change hardcoded css to use variables on animation

KevinDavilaDotCMS commented on core-web/libs/portlets/dot-content-drive/portlet/src/lib/components/dialogs/dot-content-drive-dialog-folder/dot-content-drive-dialog-folder.component.scss:
```diff
@@ -0,0 +1,113 @@
+        .p-tabview-panels {
+            position: relative;
+            overflow: hidden;
+            transition: max-height 0.4s cubic-bezier(0.4, 0, 0.2, 1);
```
Done!

KevinDavilaDotCMS commented on core-web/libs/portlets/dot-content-drive/portlet/src/lib/components/dialogs/dot-content-drive-dialog-folder/dot-content-drive-dialog-folder.component.ts:
```diff
@@ -0,0 +1,216 @@
+        this.folderForm.get('url')?.setValue(titleSlug);
+    });
+
+    async createFolder() {
```
Lol I forget to bring back to observables haha
Changed!

kevindaviladev committed:
Moved setupResizeObserverMock to test-setup lib file

KevinDavilaDotCMS commented on core-web/libs/portlets/dot-content-drive/portlet/src/lib/dot-content-drive-shell/dot-content-drive-shell.component.spec.ts:
```diff
@@ -37,6 +39,10 @@ import {
 
+// We need to setup the ResizeObserver mock globally for testing
+// PrimeNG Tabs use this and fails if not setup
+setupResizeObserverMock();
```
Moved!

kevindaviladev committed:
Changed name of allowed file extensions const
